### PR TITLE
replaced ents.GetAll() by ents.FindByClass() and cached most of the colors

### PR DESCRIPTION
**Pull request by ExtReMLapin**
ents.FindByClass() is MUCH MUCH faster than ents.GetAll with class check

Also cached most of the colors, reducing the ram stress, allocator stress and garbage collector stress

### CODE REVIEW

File: upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua [hunk 1]
```diff
@@ -4033,7 +4036,7 @@ else
 				local x, y = 0, 0;
 				
 				surface.SetMaterial(Clockwork.ScreenBlur);
-				surface.SetDrawColor(255, 255, 255, 255);
```
Comment (kurozael) [inline upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua]: This should be SetDrawColor(color_white), did you do a Find and Replace or something? 😨 There are quite a few instances of this mistake in this file.

Comment (kurozael) [reply upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua]: To clarify, the mistake is that it was changed to `SetDrawcolor_white`.

Comment (ExtReMLapin) [reply upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua]: oopsie woopsie

Comment (ExtReMLapin) [reply upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua]: [Well actually -unless i'm missing something here- SetDrawColor(kshort, kshort, kshort) is faster than SetDrawColor(color)](https://i.imgur.com/kyR643Q.png)

I didn't compare the two different type of calls before, maybe I should just have stripped Color().

Comment (ExtReMLapin) [reply upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua]: whatever i'm lazy

Comment (kurozael) [reply upload/garrysmod/gamemodes/clockwork/framework/sh_kernel.lua]: That’s a shame 😥

**Pull request closed without merging**